src/main/java/in/projecteka/consentmanager/dataflow/DataFlowBroadcastListener.java

```diff
@@ -51,22 +54,54 @@ public void subscribe() throws ClientError {
                         (DataFlowRequestMessage) converter.fromMessage(message);
                 logger.info("Received message for Request id : {}", dataFlowRequestMessage
                         .getTransactionId());
-                DataFlowRequest dataFlowRequest = DataFlowRequest.builder()
-                        .transactionId(dataFlowRequestMessage.getTransactionId())
-                        .dataPushUrl(dataFlowRequestMessage.getDataFlowRequest().getDataPushUrl())
-                        .consent(dataFlowRequestMessage.getDataFlowRequest().getConsent())
-                        .dateRange(dataFlowRequestMessage.getDataFlowRequest().getDateRange())
-                        .keyMaterial(dataFlowRequestMessage.getDataFlowRequest().getKeyMaterial())
-                        .build();
-                configureAndSendDataRequestFor(dataFlowRequest);
+                String consentArtefactId = dataFlowRequestMessage.getDataFlowRequest().getConsent().getId();
+                if (consentNotExpired(consentArtefactId)) {
+                    DataFlowRequest dataFlowRequest = DataFlowRequest.builder()
+                            .transactionId(dataFlowRequestMessage.getTransactionId())
+                            .dataPushUrl(dataFlowRequestMessage.getDataFlowRequest().getDataPushUrl())
+                            .consent(dataFlowRequestMessage.getDataFlowRequest().getConsent())
+                            .dateRange(dataFlowRequestMessage.getDataFlowRequest().getDateRange())
+                            .keyMaterial(dataFlowRequestMessage.getDataFlowRequest().getKeyMaterial())
+                            .build();
+                    configureAndSendDataRequestFor(dataFlowRequest);
```

Comment:
It should happen when the data flow request was received.

Not sure about doing it here additionally, as HIP should do the expiry validation throw an error.

In future, CM needs to send a notification if there was (any) failure response from HIP.

Reply:
Date/timestamp checks are done at the last leg of the execution.